python/search_widget/search_widget.py

```diff
@@ -0,0 +1,150 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+A simple search field with magnifying glass icon, hint text and clear button
+"""
+
+import sgtk
+from sgtk.platform.qt import QtCore, QtGui
+
+from .ui.search_widget import Ui_SearchWidget
+
+class SearchWidget(QtGui.QWidget):
+    """
+    Search widget class
+    """
+    
+    # emited when the search QTextField is being edited
+    search_edited = QtCore.Signal(object)# search text
```

Comment:
Why `object` and not `str`? I am sure there is a good reason but a small comment explaining would be cool!